src/client/Microsoft.Identity.Client/TokenCache.ITokenCacheInternal.cs

```diff
@@ -1232,6 +1307,20 @@ bool ITokenCacheInternal.HasTokensNoLocks()
             return Accessor.HasAccessOrRefreshTokens();
         }
 
+        internal /* internal for test only */ void RemoveOboTokensInternal(string partitionKey, RequestContext requestContext)
+        {
+            ILoggerAdapter logger = requestContext.Logger;
+
+            //Filter and remove tokens based on OBO Cache Key
+            var refreshTokens = Accessor.GetAllRefreshTokens(partitionKey);
+            refreshTokens.RemoveAll(item => !item.OboCacheKey.Equals(partitionKey, StringComparison.OrdinalIgnoreCase));
+            RemoveRefreshTokens(refreshTokens, logger, out bool filterByClientId);
+
+            var accessTokens = Accessor.GetAllAccessTokens(partitionKey);
+            accessTokens.RemoveAll(item => !item.OboCacheKey.Equals(partitionKey, StringComparison.OrdinalIgnoreCase));
+            RemoveAccessTokens(accessTokens, logger, filterByClientId);
+        }
+
         internal /* internal for test only */ void RemoveAccountInternal(IAccount account, RequestContext requestContext)
```

Comment:
Why do we need to update this method?

Reply:
I extracted the token removal logic to from the method to reuse the logic for the new `RemoveOboTokensInternal` method. If I didnt do this, there would be a bunch of duplicate code. There are minor differences between the two methods because the filtering logic is slightly different. calling the new remove obo api will not call `RemoveAccountInternal` so no duplicate work is done.